homeworks/lesia.turua_lesiaturuta/PokemonProject/scripts/pokemon.js

```diff
@@ -309,58 +66,138 @@ const renderCards = (cardContainer, pokemonsArray) => {
+    const types = [];
+    array.forEach((pokemon) => pokemon.type.forEach((type) => {
+        if (!types.includes(type)) types.push(type);
+    }));
+    return types;
+};
+
+const getPokemonsByType = (type, arrayPokemons) => arrayPokemons
+    .filter((pokemon) => pokemon.type
+        .includes(type));
+const getPokemonsByValet = (text, arrayPokemons) => arrayPokemons
+    .filter((pokemon) => pokemon.name
+        .toLocaleLowerCase()
+        .includes(text.toLocaleLowerCase()));
+
+const getPokemons = async (urlPokemons) => {
+    try {
+        const response = await fetch(urlPokemons);
+        return await response.json();
+    } catch (e) {
+        alert('Error url, please update this page');
+        console.error(e);
+        return [];
     }
 };
 
-const filterPokemons = (checkBig, checkSmall) => {
-    let newPokemons = pokemons;
-    if (checkBig && checkSmall) {
-        newPokemons = pokemons.filter((pokemon) => pokemon.weight < 100 && pokemon.weight > 50);
-    } else if (checkBig && !checkSmall) {
-        newPokemons = pokemons.filter((pokemon) => pokemon.weight > 100);
-    } else if (!checkBig && checkSmall) {
-        newPokemons = pokemons.filter((pokemon) => pokemon.weight < 50);
+const getFilterPokemons = () => {
+    let newPokemons = [];
+    if (searchValue && typeValue) {
+        newPokemons = getPokemonsByValet(searchValue, pokemons);
+        newPokemons = newPokemons.length
+            ? getPokemonsByType(typeValue, newPokemons)
+            : getPokemonsByType(typeValue, pokemons);
+    } else if (searchValue && !typeValue) {
+        newPokemons = getPokemonsByValet(searchValue, pokemons);
+    } else if (!searchValue && typeValue) {
+        newPokemons = getPokemonsByType(typeValue, pokemons);
+    } else if (!typeValue && !searchValue) {
+        newPokemons = pokemons;
+    }
+
+    if (checkboxBig && checkboxSmall && newPokemons.length) {
+        newPokemons = newPokemons.filter((pokemon) => pokemon.weight < 100 && pokemon.weight > 50);
+    } else if (checkboxBig && !checkboxSmall && newPokemons.length) {
+        newPokemons = newPokemons.filter((pokemon) => pokemon.weight > 100);
+    } else if (!checkboxBig && checkboxSmall && newPokemons.length) {
+        newPokemons = newPokemons.filter((pokemon) => pokemon.weight < 50);
+    }
+
+    if (highFilterEl.checked && newPokemons.length) {
+        newPokemons = sortPokemonsDown(newPokemons);
+    }
+    if (lowFilterEl.checked && newPokemons.length) {
+        newPokemons = sortPokemonsUp(newPokemons);
     }
+    typesPokemons = updateTypes(newPokemons);
+    renderSelect(typeValue);
     return newPokemons;
 };
 
-const init = () => {
-// disabled for html elements
-    selectSearchEl.disabled = true;
-    selectSearchRadioOnlyNewEl.disabled = true;
-    selectSearchRadioFavoritesEl.disabled = true;
-    // selectSearchInputSearchEl.disabled = true;
-    // selectSearchInputButtonEl.disabled = true;
+const init = async () => {
+    try {
```

Comment:
If you have `try catch ` block in the `getPokemons` function, you can skip it here

Reply:
fixed